src/init.cpp

```diff
@@ -1110,8 +1106,12 @@ bool AppInit2()
     }
 
 
+    uiInterface.InitMessage(_("Loading Network Averages..."));
+    if (fDebug3) printf("Loading network averages");
     if (!NewThread(StartNode, NULL))
         InitError(_("Error: could not start node"));
+    else
+        BusyWaitForTally();
```

Comment:
Putting else here is probably pointelss. `NewThread` will only fail when OS runs out of memory / PIDs, and in that case app will probably exit from `InitError`.